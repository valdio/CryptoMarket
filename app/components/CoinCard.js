import React from 'react'
import {View, Text, StyleSheet} from 'react-native'
import {COLORS, Spacing, Typography} from '../template'
import FastImage from 'react-native-fast-image'

export default CoinCard = ({metadata} = props) => {
  return (<View style={styles.container}>
    <View style={styles.iconContainer}>
      <FastImage
        testID="card-coinIcon"
        style={styles.coinIcon}
        source={{uri: metadata.logo || '', priority: FastImage.priority.normal}}
        resizeMode={FastImage.resizeMode.contain}
      />
    </View>
    <View style={styles.content}>
      <Text style={styles.cardTitle} testID="card-title">{metadata.name}</Text>
      <Text style={styles.description} testID="card-description">{metadata.description}</Text>
      <View style={styles.tagsHolder}>
        {metadata.tags && metadata.tags.map(tag => <Text key={tag} style={styles.tag}>{tag}</Text>)}
      </View>
    </View>
  </View>)
}

export const styles = StyleSheet.create({
  container: {
    ...Spacing.horizontalSpacing,
    borderRadius: 5,
    backgroundColor: COLORS.WHITE
  },
  content: {
    ...Spacing.section,
    alignItems: 'flex-start'
  },
  iconContainer: {
    backgroundColor: COLORS.WHITE,
    height: 64,
    width: 64,
    borderRadius: 32,
    borderWidth: 2,
    alignItems: 'center',
    borderColor: COLORS.WHITE,
    position: 'absolute',
    marginTop: -32,
    marginHorizontal: 10
  },
  coinIcon: {
    height: 60,
    width: 60
  },
  cardTitle: {
    ...Typography.boldText,
    color: COLORS.DARK_TEXT_COLOR,
    fontSize: 20,
    marginBottom: 6
  },
  description: {
    fontSize: 14,
    color: COLORS.FADED_TEXT_COLOR
  },
  tagsHolder: {
    alignSelf: 'flex-end',
    flexDirection: 'row',
    justifyContent: 'flex-end',
    marginTop: 12,
    marginBottom: 6
  },
  tag: {
    ...Typography.boldText,
    color: COLORS.GREEN,
    marginHorizontal: 5,
    lineHeight: 17
  }
})

