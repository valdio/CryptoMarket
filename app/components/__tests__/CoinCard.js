import React from 'react'
import renderer from 'react-test-renderer'
import CoinCard from '../CoinCard'
import {cryptoMetadataMock} from './mock'
import {shallow} from 'enzyme'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import {Text, View} from 'react-native'
import {styles} from '../CoinCard'
import FastImage from 'react-native-fast-image'

Enzyme.configure({adapter: new Adapter()})
const wrapper = shallow(<CoinCard metadata={cryptoMetadataMock}/>)

it('renders CoinCard using Snapshots', () => {
  const tree = renderer.create(<CoinCard metadata={cryptoMetadataMock}/>)
  expect(tree).toMatchSnapshot()
})

test('CoinCard renders title', () => {
  expect(wrapper.contains(<Text style={styles.cardTitle}
                                testID="card-title">{cryptoMetadataMock.name}</Text>)).toBe(true)
})

test('CoinCard has Text components: title, description & crypto-tag', () => {
  expect(wrapper.find(Text).length).toBe(3)
})

describe('CoinCard has cryptocurrency icon', () => {
  expect(wrapper.find(FastImage).length).toBe(1)
})

describe('CoinCard has title', () => {
  expect(wrapper.findWhere(node => node.prop('testID') === 'card-title').text()).toBeTruthy()
})

describe('CoinCard has description', () => {
  expect(wrapper.findWhere(node => node.prop('testID') === 'card-description').text()).toBeTruthy()
})
