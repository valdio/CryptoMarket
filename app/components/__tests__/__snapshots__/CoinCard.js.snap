// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CoinCard using Snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "marginHorizontal": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FFFFFF",
        "borderRadius": 32,
        "borderWidth": 2,
        "height": 64,
        "marginHorizontal": 10,
        "marginTop": -32,
        "position": "absolute",
        "width": 64,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 60,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "priority": "normal",
            "uri": "https://s2.coinmarketcap.com/static/img/coins/64x64/1.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "marginHorizontal": 20,
        "marginVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#484848",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 6,
        }
      }
      testID="card-title"
    >
      Bitcoin
    </Text>
    <Text
      style={
        Object {
          "color": "#8193AE",
          "fontSize": 14,
        }
      }
      testID="card-description"
    >
      Bitcoin (BTC) is a consensus network that enables a new payment system and a completely digital currency. Powered by its users, it is a peer to peer payment network that requires no central authority to operate. On October 31st, 2008, an individual or group of individuals operating under the pseudonym "Satoshi Nakamoto" published the Bitcoin Whitepaper and described it as: "a purely peer-to-peer version of electronic cash, which would allow online payments to be sent directly from one party to another without going through a financial institution."
    </Text>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginBottom": 6,
          "marginTop": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#56CB8F",
            "fontSize": 14,
            "fontWeight": "bold",
            "lineHeight": 17,
            "marginHorizontal": 5,
          }
        }
      >
        mineable
      </Text>
    </View>
  </View>
</View>
`;
